graf2d/primitives/v7/inc/ROOT/RLine.hxx

```diff
@@ -83,6 +83,50 @@ GetDrawable(const std::shared_ptr<ROOT::Experimental::RLine> &line)
    return line;
 }
 
+
+class RLineNew : public RDrawable {
+
+   RPadPos fP1;           ///< 1st point
+   RPadPos fP2;           ///< 2nd point
+
+   RDrawingOptsBase fOpts; ///<! only temporary here, should be removed later
+
+   RDrawableAttributes fAttr{"line"}; ///< attributes
+
+public:
+
+   RLineNew() = default;
+
+   RLineNew(const RPadPos& p1, const RPadPos& p2) : fP1(p1), fP2(p2) {}
+
+   void SetP1(const RPadPos& p1) { fP1 = p1; }
+   void SetP2(const RPadPos& p2) { fP2 = p2; }
+
+   const RPadPos& GetP1() const { return fP1; }
+   const RPadPos& GetP2() const { return fP2; }
+
+   RAttrLineNew AttrLine() { return RAttrLineNew(fAttr); }
```

Comment:
This seems to return a copy - but IIUC the returned type behaves like a reference. That breaks value semantics that we were hoping to retain...

Reply:
Sorry, this is just copy-and-paste from `RLine` class. 
I did not touch `RPadPos` / `RPadLength` classes, but I have an idea how they can be re-implemented with `RAttributesVisitor` class. While any attribute in RAttributesContainer is optional, one can very easily solve problem with superposition of different coordinates system - **pixel**/**relative**/**user**/**anything_else_system**